fhir-server/src/main/java/com/ibm/fhir/server/filter/rest/FHIRRestServletFilter.java

```diff
@@ -261,6 +274,50 @@ private HTTPReturnPreference computeReturnPref(ServletRequest request, HTTPHandl
         return returnPref;
     }
 
+    /**
+     * Check that FHIR version parameters in the HTTP headers are valid.
+     * @param requestHeaders the headers
+     * @throws FHIRRestServletRequestException
+     */
+    void checkFhirVersionParameter(Map<String, List<String>> requestHeaders) throws FHIRRestServletRequestException {
+        Map<String, Integer> headerStatusMap = new LinkedHashMap<>();
+        // 415 Unsupported Media Type is the appropriate response when the client posts a format that is not supported to the server.
+        headerStatusMap.put(HttpHeaders.CONTENT_TYPE, HttpServletResponse.SC_UNSUPPORTED_MEDIA_TYPE);
+        // 406 Not Acceptable is the appropriate response when the Accept header requests a format that the server does not support.
+        headerStatusMap.put(HttpHeaders.ACCEPT, HttpServletResponse.SC_NOT_ACCEPTABLE);
+
+        for (String headerName : headerStatusMap.keySet()) {
+            String fhirVersion = null;
+            for (String headerValue : requestHeaders.getOrDefault(headerName, Collections.emptyList())) {
+                if (headerValue != null) {
+                    for (String headerValueElement : headerValue.split(",")) {
+                        Map<String, String> parameters = MediaType.valueOf(headerValueElement).getParameters();
+                        if (parameters != null) {
```

Comment:
can parameters actually be null.  from https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/MediaType.html#getParameters-- I think not, but probably needs a quick test to be sure

Reply:
at least for the version in my workspace, I still think not:
```
    private MediaType(String type, String subtype, String charset, Map<String, String> parameterMap) {

        this.type = type == null ? MEDIA_TYPE_WILDCARD : type;
        this.subtype = subtype == null ? MEDIA_TYPE_WILDCARD : subtype;

        if (parameterMap == null) {
            parameterMap = new TreeMap<String, String>(new Comparator<String>() {

                @Override
                public int compare(String o1, String o2) {
                    return o1.compareToIgnoreCase(o2);
                }
            });
        }

        if (charset != null && !charset.isEmpty()) {
            parameterMap.put(CHARSET_PARAMETER, charset);
        }
        this.parameters = Collections.unmodifiableMap(parameterMap);
    }
```

Reply:
Ah, right. I didn't verify that all constructors indeed set the parameters member variable. Looks like they all do.